_posts/2020-08-14-how-to-stay-sane-with-robot-framework.md

```diff
@@ -0,0 +1,252 @@
+---
+layout: post
+title: How to stay sane with robot framework and plaa plaa
+author: Tuomo Turunen
+excerpt: >
+  Robot framework with Selenium offers a wide toolbox to work on with
+  browser UI tests. With simple patterns and architechture, writing and 
+  maintaining UI tests is not a big deal as it used to be.
+tags:
+ - Robot framework
+ - Test automation
+---
+
+How to stay sane with web application test automation with the Robot Framework 
+
+Intro
+-----
+My first experience in Selenium was in university approx 15 years ago and it was an optional part of a bigger assignment. Wasn’t surprised that only a few people managed to complete it. Back in the days, writing browser UI tests were more and less black magic: keywords were high level yet translating low-level technical requirements into selenium keywords had a steep learning curve. After plus ten years of professional software development, I had a Deja Vu moment where there was an optional assignment and it involved Selenium with a twist of robot framework. 
+
+Compared to my previous experience with browser UI testing, the documentation and usage were much more userfriendly, and getting started didn’t include frustrating and complex setup. Over four months there were hard times when not everything worked as I thought yet with some trial and error I managed to achieve project goals without sacrificing too much of my sanity. Like always in software engineering, the winning receipt was to find common patterns and general solutions for 80% of requirements and avoid using duct tape solutions.
+
+When designing UI tests, test fragility is the key thing to keep in mind. When the UI changes, how many things I have to change in my code? How can I uniquely click objects when app has dynamic content and in a shared environment? Simple patterns lead to simpler solutions that are more maintainable and easier to comprehend when the project is transferred to another person and hopefully this blog post gives you an idea or two how to leave a solid project for the other person to continue.
+
```

Comment:
I think this was a good intro, selling the idea well why to read further, common pains with testing as well. 
when the app